go-controller/pkg/ovn/controller/unidling/unidle.go

```diff
@@ -0,0 +1,138 @@
+package unidling
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+
+	"k8s.io/klog/v2"
+)
+
+// unidlinController checks periodically the OVN events db
+// and generates a Kubernetes NeedPods events with the Service
+// associated to the VIP
+type unidlingController struct {
+	eventRecorder record.EventRecorder
+	// Map of load balancers to service namespace
+	serviceVIPToName     map[ServiceVIPKey]types.NamespacedName
```

Comment:
consider not using a struct here as @girishmg pointed out it will consume more memory. Perhaps we could just use a string of "<vip>:<protocol>"?

Reply:
before the services were never deleted, so the map was unbounded and always growing, this is 1:1 to the number of services, this is not am scalability problem

Reply:
check the memory usage using a string vs a struct as key with 10k services, the difference is not significative

```
go test -bench=. -v
goos: linux
goarch: amd64
BenchmarkStruct
BenchmarkStruct-4   	    3740	    343477 ns/op
BenchmarkString
BenchmarkString-4   	    3475	    316909 ns/op
PASS
ok  	_/home/aojea/tmp	2.464s
```

these is the code
```go
package main

import (
	"fmt"
	"testing"
)

// ServiceVIPKey is used for looking up service namespace information for a
// particular load balancer
type ServiceVIPKey struct {
	// Load balancer VIP in the form "ip:port"
	vip string
	// Protocol used by the load balancer
	protocol string
}

var x map[ServiceVIPKey]string
var y map[string]string

func main() {

	fmt.Println("Struct   ", testing.Benchmark(BenchmarkStruct))
	fmt.Println("String   ", testing.Benchmark(BenchmarkString))
}

func BenchmarkStruct(b *testing.B) {
	for i := 0; i < b.N; i++ {
		x = make(map[ServiceVIPKey]string, 10000)
	}
}

func BenchmarkString(b *testing.B) {
	for i := 0; i < b.N; i++ {
		y = make(map[string]string, 10000)
	}
}
```

Reply:
thanks @aojea